api/tests/Feature/PoolCandidateUpdateTest.php

```diff
@@ -1229,4 +1229,143 @@ public function testManualStatusUpdatePolicy(): void
                 ],
             ]);
     }
+
+    /**
+     * @dataProvider nullTimeProvider
+     */
+    public function testNullTimestampsOnStatusChanged($status, $expected, $user)
+    {
+        $past = config('constants.past_datetime');
+        $this->poolCandidate->pool_candidate_status = PoolCandidateStatus::DRAFT->name;
+        $this->poolCandidate->submitted_at = $past;
+        $this->poolCandidate->removed_at = $past;
+        $this->poolCandidate->final_decision_at = $past;
+        $this->poolCandidate->placed_at = $past;
+        $this->poolCandidate->save();
+
+        $this->actingAs($this->$user, 'api')
+            ->graphQL($this->manualStatusUpdateMutation, [
+                'id' => $this->poolCandidate->id,
+                'candidate' => ['status' => $status],
+            ])->assertJsonFragment([
+                'status' => [
+                    'value' => $status,
+                ],
+                ...$expected,
+            ]);
+    }
+
+    public static function nullTimeProvider()
+    {
+        $nullState = [
+            'removedAt' => null,
+            'finalDecisionAt' => null,
+            'placedAt' => null,
+        ];
+
+        $finalDeciscion = [
```

Comment:
```suggestion
        $finalDecision = [
```

Reply:
Oops, fixed in [fix tests](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12734/commits/85a495b5af47cef5016356d9676659f6d599289f)